stream-platform/src/main/java/io/camunda/zeebe/stream/impl/BufferedTaskResultBuilder.java

```diff
@@ -26,25 +27,32 @@
 final class BufferedTaskResultBuilder implements TaskResultBuilder {
 
   private final MutableRecordBatch mutableRecordBatch;
+  private final StagedScheduledCommandCache cache;
 
-  BufferedTaskResultBuilder(final RecordBatchSizePredicate predicate) {
+  BufferedTaskResultBuilder(
+      final RecordBatchSizePredicate predicate, final StagedScheduledCommandCache cache) {
     mutableRecordBatch = new RecordBatch(predicate);
+    this.cache = cache;
   }
 
   @Override
   public boolean appendCommandRecord(
       final long key, final Intent intent, final UnifiedRecordValue value) {
-
     final ValueType valueType = TypedEventRegistry.TYPE_REGISTRY.get(value.getClass());
     if (valueType == null) {
       // usually happens when the record is not registered at the TypedStreamEnvironment
       throw new IllegalStateException("Missing value type mapping for record: " + value.getClass());
     }
 
+    if (cache.contains(intent, key)) {
```

Comment:
Nice! This would even mean that we might be able to boost performance here if we had multiple timers before which we had written and part of the batch, they are even not in the batch, meaning more we can add to the batch (which are likely the ones we haven;t written yet.) Nice.